script.js

```diff
@@ -0,0 +1,487 @@
+      textInput.selectionEnd = cursorPosition;
+    }
+  },
+  leftArrow() {
+    const cursorPosition = textInput.selectionEnd;
+    if (cursorPosition === 0) {
+      return;
+    }
+    textInput.setSelectionRange(cursorPosition - 1, cursorPosition - 1);
+  },
+  rightArrow() {
+    const cursorPosition = textInput.selectionStart;
+    if (cursorPosition < textInput.value.length) {
+      textInput.setSelectionRange(cursorPosition + 1, cursorPosition + 1);
+    }
+  },
+  capslock(evt) {
+    if (evt.detail.repeated) {
+      return;
+    }
+    isCapsOn = !isCapsOn;
+    capslockKey.classList.toggle('caps-active');
+    keyElementsList.forEach((elem, index) => {
+      const keyElem = elem;
+      if (isRegularLetter(keyElem.innerText)) {
+        keyElem.innerText = isCapsOn
+          ? keyElem.innerText.toUpperCase()
+          : keyElem.innerText.toLowerCase();
+      }
+      if (isMultiContentKey(index)) {
+        keyElem.innerText = isCapsOn ? lettersList[`${lang}Keys`][index][1] : lettersList[`${lang}Keys`][index][0];
+      }
+    });
+  },
+  onShiftUppercase(evt) {
+    if (evt.type === 'mousedown' && !evt.detail.repeated) {
+      keyElementsList.forEach((elem) => {
+        const keyElem = elem;
+        if (isRegularLetter(keyElem.innerText)) {
+          keyElem.innerText = keyElem.innerText.toUpperCase();
+        }
+      });
+    }
+  },
+  onShiftLowercase() {
+    if (isCapsOn) {
+      return;
+    }
+    keyElementsList.forEach((elem) => {
+      const keyElem = elem;
+      if (isRegularLetter(keyElem.innerText)) {
+        keyElem.innerText = keyElem.innerText.toLowerCase();
+      }
+    });
+  },
+  onLangChange() {
+    clearTimeout(timer);
+    specialKeysHandlers.onShiftLowercase();
+    // }
+    lang = (lang === 'eng') ? 'rus' : 'eng';
+
+    if (isStorageSupport) {
+      localStorage.setItem('lang', lang);
+    }
+
+    keyElementsList.forEach((elem, index) => {
+      const keyElem = elem;
+      const letter = lettersList[`${lang}Keys`][index];
+      if (isMultiContentKey(index)) {
+        keyElem.innerText = isCapsOn ? letter[1] : letter[0];
+        return;
+      }
+      if (isRegularLetter(keyElem.innerText)) {
+        keyElem.innerText = isCapsOn ? letter.toUpperCase() : letter.toLowerCase();
+      }
+    });
+  },
+};
+
+function addHandlers() {
+  keyboardElem.addEventListener('mousedown', (evt) => {
+    const targetKey = evt.target;
+
+    if (!targetKey.classList.contains('key')) {
+      return;
+    }
+
+    if (!evt.detail.repeated) {
+      pressedKeysList.push(targetKey);
+      targetKey.classList.add('active');
+    }
+
+    const innerText = (evt.target.innerText === '') ? ' ' : evt.target.innerText;
+
+    if (isRegularLetter(innerText)) {
+      textTyping(innerText);
+    }
+
+    if (pressedKeysList.length > 1
+      && isLangChangingPressed(pressedKeysList)
+      && !evt.detail.repeated) {
+      specialKeysHandlers.onLangChange();
+    }
```

Comment:
Very hard to read this conditions, move it to separated variable and use it, please.